src/index.tsx

```diff
@@ -15,11 +15,53 @@ import React, {
 // eslint-disable-next-line @typescript-eslint/ban-types
 type ComponentDidCatch = ComponentLifecycle<{}, {}>["componentDidCatch"];
 
+type ErrorType = WrappedError | Error | undefined;
+
 interface ErrorBoundaryProps {
-  error: unknown;
+  error: ErrorType;
   onError: NonNullable<ComponentDidCatch>;
 }
 
+/**
+ * Wrapper around the `Error` class to allow us to create an error
+ * message while retaining the originally thrown data as an attribute
+ * on the class.
+ */
```

Comment:
```suggestion
/**
 * Wrapper that is instantiated for thrown primitives so that consumers always work with the `Error` interface.
 * The thrown primitive can be accessed via the `originalData` property.
 */
```